File: word2json/models/article.py
# %%
from typing import Optional
from word2json.models.base import *


class Article(BaseDocument):
    articleUrls: Optional[list[ArticleUrl]]
    author: Optional[str]
    body: Optional[list[Body]]
    category: Optional[str]
    description: Optional[str]
    featured: Optional[bool]
    mainImageUrl: Optional[str]
    publishedAt: Optional[str]
    slug: Optional[Slug]
    tags: Optional[list[str]]
    title: Optional[str]
    writtenAt: Optional[str]

File: json2word/models/article.ts
import { Body, Slug, BaseDocument, ArticleUrl } from "./base";

export interface Article extends BaseDocument {
	articleUrls?: ArticleUrl[];
	author?: string;
	body?: Body[];
	category?: string;
	description?: string;
	featured?: boolean;
	mainImageUrl?: string;
	publishedAt: string;
	writtenAt: string;
	slug: Slug;
	tags?: string[];
	title: string;
}
